### DISCUSSION TITLE
[Question] Permission denied on mobile

### DISCUSSION THREAD

**Discussion by Vallekralle** [flet-dev/flet#2541] [Q&A] [answered]
### Question

I tried to write to a new file that doesn't exists yet (on mobile). With the code snippet: 

```
video_req = requests.get(url=self.data[0]['urls'][1]["url"], allow_redirects=True)  # just gets the video with an api, it works fine

with open(f"/storage/emulated/0/Download/literallyme.mp4", "wb") as f:
    f.write(video_req.content)
```

When I try this on my pc it works perfectly fine, without any problem, but when I build the apk and install the app on my mobile phone, it shows the error msg: `[Errno 13] Permission denied: '/storage/emulated/0/Download/literallyme.mp4'`. I think my problem is that I need the permission, like when you try to get a file or folder with the FilePicker-Class from flet. I automatically shows an AlertDialog with the permission inquiry.

I hope you can help me that would be great 👍 

Thank you for reading my question/problem,
Stay safe and have fun programming

### Code sample

```python
video_req = requests.get(url=self.data[0]['urls'][1]["url"], allow_redirects=True)  # just gets the video with an api, it works fine

with open(f"/storage/emulated/0/Download/literallyme.mp4", "wb") as f:
    f.write(video_req.content)
```

### Error message

```bash
[Errno 13] Permission denied: '/storage/emulated/0/Download/literallyme.mp4'
```

### ------------------------------------------------------

- [x] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by burhansvural** [comment, marked as answer]
> ### Question
> I tried to write to a new file that doesn't exists yet (on mobile). With the code snippet:
> 
> ```
> video_req = requests.get(url=self.data[0]['urls'][1]["url"], allow_redirects=True)  # just gets the video with an api, it works fine
> 
> with open(f"/storage/emulated/0/Download/literallyme.mp4", "wb") as f:
>     f.write(video_req.content)
> ```
> 
> When I try this on my pc it works perfectly fine, without any problem, but when I build the apk and install the app on my mobile phone, it shows the error msg: `[Errno 13] Permission denied: '/storage/emulated/0/Download/literallyme.mp4'`. I think my problem is that I need the permission, like when you try to get a file or folder with the FilePicker-Class from flet. I automatically shows an AlertDialog with the permission inquiry.
> 
> I hope you can help me that would be great 👍
> 
> Thank you for reading my question/problem, Stay safe and have fun programming
> 
> ### Code sample
> ```python
> video_req = requests.get(url=self.data[0]['urls'][1]["url"], allow_redirects=True)  # just gets the video with an api, it works fine
> 
> with open(f"/storage/emulated/0/Download/literallyme.mp4", "wb") as f:
>     f.write(video_req.content)
> ```
> 
> ### Error message
> ```shell
> [Errno 13] Permission denied: '/storage/emulated/0/Download/literallyme.mp4'
> ```
> 
> ### ------------------------------------------------------
> * [x]  I have searched for answers to my question both in the issues and in previous discussions.

https://github.com/kivy/python-for-android/blob/develop/pythonforandroid/recipes/android/src/android/permissions.py

**Reply by Vallekralle** [reply]
Thank you for your reply 👍
